benchmark_runner/jupyterlab/templates/analyze_workloads/analyze_workloads_operations.py

```diff
@@ -0,0 +1,129 @@
+
+# display df
+from IPython.display import clear_output
+
+# display bokeh
+from bokeh.plotting import figure, show
+from bokeh.io import output_notebook, output_file
+from bokeh.models import HoverTool, Label, ColumnDataSource, LabelSet, Circle, NumeralTickFormatter
+
+
+class AnalyzeWorkloadOperations:
+    """
+    This class for analyze workloads operations
+    """
+
+    def __init__(self):
+        self.clear_output()
+
+    @staticmethod
+    def clear_output():
+        """
+        This method clean output
+        @return:
+        """
+        # clean latest bokeh output
+        clear_output(wait=True)
+
+    @staticmethod
+    def compare_results(result1: dict, result2: dict, legend_label1: str, legend_label2: str, **workload_data: dict):
+        """
+        This method compare between 2 results
+        @param result1: thread results in dictionary
+        @param result2: thread results in dictionary
+        @param legend_label1: first bar legend
+        @param legend_label2: second bar legend
+        @return:
+        """
+        title = workload_data.get('title')
+        x_axis_label = workload_data.get('x_axis_label')
+        y_axis_label = workload_data.get('y_axis_label')
+
+        # Convert values to integers
+        # Convert values to integers
+        values1 = [float(value) for value in result1.values()]
+        values2 = [float(value) for value in result2.values()]
+
+        metrics = list(result1.keys())
+
+        width = 0.3
+        x = [i for i in range(len(metrics))]
+        offset = 0.1  # Adjust the offset for x-axis shift
+        x_middle = [i + width / 2 + offset for i in x]  # Adjust x-coordinates for centering and shift
+
+        p = figure(title=title, x_range=metrics, width=1200, height=600)
```

Comment:
declare width and height as const as these are fixed size.

Reply:
Fixed